src/components/interface/_utils/useFestivalData.tsx

```diff
@@ -1,14 +1,31 @@
 import { useEffect, useState } from "react";
-import { Festival } from "../../../types";
-import { getFestivalData } from "../../../database";
+import { RawFestival } from "../../../types";
+import { supabaseMethods } from "../../../database/supabase";
 
 export const useFestivalData = () => {
-  const [festivals, setFestivals] = useState<Festival[]>([]);
+  const [festivals, setFestivals] = useState<RawFestival[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    const festivals = getFestivalData();
+  const fetchData = async () => {
+    setIsLoading(true);
+    const festivals = await getFestivalData();
+    setIsLoading(false);
     setFestivals(festivals);
+  };
+
+  useEffect(() => {
+    fetchData();
   }, []);
 
-  return festivals;
+  return { festivals, isDataFetching: isLoading };
+};
+
+const getFestivalData = async (): Promise<RawFestival[]> => {
+  try {
+    const { data: festivals } = await supabaseMethods.readApprovedFestivals();
+    if (!festivals) throw new Error();
+    return festivals;
+  } catch {
+    return [];
+  }
```

Comment:
These changes might be the cause of the loading bug, double check that `isLoading` bool is working.

Reply:
Loading issue was related to something else.